### ISSUE TITLE
I have a question.

### ISSUE THREAD

**Issue by Wml333** [yandex/odyssey#40]
Hi! I am a student.Recently,I am studing odyssey.After reading the source code, I would like to ask you a question!The main role of the odysey worker thread is as follows. I don't know if it is right. The client sends a request to the console. The console sends a message type to the worker thread by creating a message channel, the worker thread matches the route, and then the route matches the database server. Finally, the database server matches the data to the worker thread through the route, and the worker thread passes. The console replies to the client. Therefore, when the number of client accesses is too large, the workload of a worker is too large, so it can be solved by adding a worker.

**Comment by pmwkaa** [comment]
Hi. Its probably a little bit trickier.

When client connects, its connection accepted from system worker: https://github.com/yandex/odyssey/blob/master/sources/system.c#L65 Then this connection transfered to a next worker from the worker pool.

Worker reads startup packet from client and asks router (which risides in system worker) to match a route for the client: https://github.com/yandex/odyssey/blob/master/sources/frontend.c#L1035

Depending on route type (remote or local) worker handles client requests. In case of console requests (which is local) all query requests are forwarded to console subsystem (which resides in system worker too): https://github.com/yandex/odyssey/blob/master/sources/frontend.c#L495

So basically, all shared data lives in system worker context. All synchronization between workers is done by using channels and messages. No direct locking is used.

One worker can host thousands of clients and new workers can be added to handle more workload.

**Comment by Wml333** [comment]
Thank you very much !

**Comment by Gorilaz** [comment]
Hey, guys! I have a question) How to reload configuration without breaking connections with clients. I mean I changed configurations in odyssey.conf and I want to apply it, but the clients remain connected to database

**Comment by reshke** [comment]
kill -s HUPSent from my Galaxy
-------- Original message --------From: Sergei <notifications@github.com> Date: 10/12/2020  19:34  (GMT+05:00) To: yandex/odyssey <odyssey@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: Re: [yandex/odyssey] I have a question. (#40) 
Hey, guys! I have a question) How to reload configuration without breaking connections with clients. I mean I changed configurations in odyssey.conf and I want to apply it, but the clients remain connected to database

**Comment by Gorilaz** [comment]
kill -HUP pid
kill -1 pid
after that we have a new pid. And if you had a connect in psql you get -
 server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Succeeded.
I don't know will it affect web sevice or not. I'll tell you later. When I use pgbouncer I don't receive that error

**Comment by Gorilaz** [comment]
Hello! Could you give some explanations about logs in odyssey.
We have situation when django exceeds all connection in pool odyssey

104 2021-05-19T18:30:46Z info [cacbec16ca527 sdb33781ecdf6] (main) client disconnected (read/write error, addr 10.10.30.38:43608): Resource temporarily unavailable, status OD_ECLIENT_READ
......
......
104 2021-05-19T18:30:47Z info [cacbec16ca527 sdb33781ecdf6] (reset) not responded, cancel (#0)
104 2021-05-19T18:30:47Z info [none none] (cancel) cancel for sdb33781ecdf6
.....
......
104 2021-05-19T18:30:48Z error [cacbec16ca527 sdb33781ecdf6] (reset) server cancel limit reached, closing
104 2021-05-19T18:30:48Z info [cacbec16ca527 sdb33781ecdf6] (main) reset unsuccessful, closing server connection

conf file odyssey
database ".." {
    user ".." {
      authentication "md5"
      storage "postgres_server"
      pool "transaction"
      pool_size 130
      pool_ttl 0
      pool_discard yes
      pool_cancel yes
      pool_rollback yes
      server_lifetime 3600
      client_fwd_error no
}
